### PR TITLE
cairo-native-test binary

### LINKED ISSUE #442: cairo-native-test: A binary to run tests within a cairo program
The cairo implementation is here <https://github.com/starkware-libs/cairo/blob/main/crates/cairo-lang-test-runner/src/lib.rs> we could copy this and make a binary/library to run functions marked as tests from a given .cairo file and extend further with tests from repositories we find on the ecosystem.

### PR DESCRIPTION
**Pull request by edg-l**
Fixes #442

### CODE REVIEW

**Review by azteca1998 — DISMISSED** [review]

**Review by juanbono — CHANGES_REQUESTED** [review]
Please add a section in the README about how to use it

**Comment by edg-l** [comment]
> Please add a section in the README about how to use it

done

File: src/executor/jit.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ impl<'m> JitNativeExecutor<'m> {
             engine: create_engine(&module, &metadata, opt_level),
             module,
             registry,
-            gas_metadata: metadata.get::<GasMetadata>().cloned(),
+            gas_metadata: metadata.get::<GasMetadata>().cloned().unwrap(),
```
Comment (juanbono) [inline src/executor/jit.rs:51]: is the GasMetadata always present? if not we shouldn't have an unwrap here .

File: src/executor/aot.rs [hunk 2]
```diff
@@ -59,25 +59,28 @@ impl AotNativeExecutor {
         Self {
             library: unsafe { Library::new(library_path).unwrap() },
             registry,
-            gas_metadata: metadata.remove(),
+            gas_metadata: metadata.remove().unwrap(),
```
Comment (juanbono) [inline src/executor/aot.rs:62]: same here

**Review by juanbono — APPROVED** [review]
LGTM

File: src/executor/jit.rs (hunk 1 above)
Comment (juanbono) [reply src/executor/jit.rs:51]: It is always inserted in src/context.rs

**Review by azteca1998 — APPROVED** [review]

**Pull request merged**